elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexTransaction.java

```diff
@@ -240,13 +240,11 @@ public void setAttribute(Object entity, String attributeName, Object attributeVa
 
     private Serializable extractId(FilterExpression filterExpression,
                                    String idFieldName,
-                                   Object parent,
-                                   Class<?> relationClass,
-                                   EntityDictionary dictionary) {
+                                   Class<?> relationClass) {
         Collection<FilterPredicate> predicates = filterExpression.accept(new PredicateExtractionVisitor());
         for (FilterPredicate predicate : predicates) {
             List<Object> values = predicate.getValues();
-            Class<?> entityClass = dictionary.getParameterizedType(parent, predicate.getField());
+            Class<?> entityClass = predicate.getEntityType();
```

Comment:
Since `Class` doesn't override `.equals`, `Class` actually falls back on `Object.equals`. i.e.:

```
public boolean equals(Object obj) {
        return (this == obj);
    }
```

So they're equivalent. If we think `.equals` is more readable we can use that. Otherwise, shouldn't make a difference :)